<?xml version="1.0"?>
<!-- Launches frame dropper and serial comms for the server side -->
<launch>
  
  <arg name="camera_1" default="/front_web"/>
  <arg name="camera_2" default="/back_web"/>
  <arg name="ip_address" default="192.168.168.11"/>
  <arg name="odom_cam_topic" default="/rgbd_odom_node/odom"/>
  
  <arg name="debug" default="false"/>
  
  <!-- Launch udp bridge -->
  
  <node pkg="udp_bridge" type="udp_client" name="udp_client" output="screen" unless="$(arg debug)">
   <param name="camera_1"  value="$(arg camera_1)"/>
   <param name="camera_2"  value="$(arg camera_2)"/>
   <param name="ip_address" value="$(arg ip_address)"/>
   <param name="odom_topic_2" value="$(arg odom_cam_topic)"/>
   <rosparam file="$(find udp_bridge)/cfg/translate_360.yaml" command="load"  /> 
 </node>
  
 <node pkg="udp_bridge" type="udp_client" name="udp_client" output="screen" launch-prefix="xterm -e gdb --args" if="$(arg debug)">
   <param name="camera_1"  value="$(arg camera_1)"/>
   <param name="camera_2"  value="$(arg camera_2)"/>
   <param name="ip_address" value="$(arg ip_address)"/>
   <param name="odom_topic_2" value="$(arg odom_cam_topic)"/>
 </node>
</launch>
